src/components/Nav/NavBar/KakaoLogin/UserInfo/UserInfo.js

```diff
@@ -0,0 +1,133 @@
+import styled from 'styled-components';
+import { useState, useRef, useEffect } from 'react';
+import LogoutBtn from '../LogoutBtn/LogoutBtn';
+import theme from '../../../../../styles/theme';
+import { useNavigate } from 'react-router';
+
+const UserInfo = ({ setToken }) => {
+  const [toggleUserMenu, setToggleUserMenu] = useState(false);
+  const menuRef = useRef();
+  const navigate = useNavigate();
+
+  const checkIfClickedOutside = e => {
+    if (
+      toggleUserMenu &&
+      menuRef.current &&
+      !menuRef.current.contains(e.target)
+    ) {
+      setToggleUserMenu(prev => !prev);
+    }
+  };
```

Comment:
```suggestion
  const checkIfClickedOutside = e => {
    const condition = toggleUserMenu && menuRef.current && !menuRef.current.contains(e.target);
    if (condition) {
      setToggleUserMenu(prev => !prev);
    }
  };
```
조건문의 조건이 긴 경우 변수화시켜서 가독성을 높여보세요!

Reply:
감사합니다 리뷰 반영했습니다 도현님 🤭